firmware/fw_base.S

```diff
@@ -116,6 +116,9 @@ _fdt_reloc_done:
 _wait_for_boot_hart:
 	la	a4, _boot_hart_done
 	REG_L	a5, (a4)
+	nop
```

Comment:
Can you please add a comment inline to describe why this is required?

Also, why can't this be a fence? That seems like a bug somewhere else if a fence doesn't work

Reply:
> Can you please add a comment inline to describe why this is required?
> 
Done.
> Also, why can't this be a fence? That seems like a bug somewhere else if a fence doesn't work

One possible explanation is non-boot cpu cache may have some stale corrupt DTB data which gets written. But this is a just theory and no way to prove it.

That's why commit text says that it's being reverted just for the time being until we figure out the actual reason.